crates/fj-window/src/run.rs

```diff
@@ -31,14 +31,25 @@ pub fn run(
     shape_processor: ShapeProcessor,
     invert_zoom: bool,
 ) -> Result<(), Error> {
+    let (send_gui, gui_event_rx) = crossbeam_channel::bounded::<GuiEvent>(5);
+
+    let (gui_event_tx, recv_gui) = crossbeam_channel::bounded::<GuiEvent>(5);
```

Comment:
Those bounded channels with capacity `5` look a bit fishy to me. Why 5? Is there some chance that these buffers can fill up and cause a deadlock?

I usually create bounded channels with capacity `0`, because if the channel *can* theoretically cause a deadlock, it usually *will* cause a deadlock if it has no buffer! If it doesn't cause a deadlock and works otherwise, it should be fine to leave it at capacity `0`, unless there are specific reasons not to (like performance).

Reply:
I changed the capacity to be 1. I am not sure why but a capacity of zero caused the GUI not to be initialized. I think the reason is that there is a single event loop and both the send and receive operations are ran on the same thread so a buffer size of 1 is necessary.

Reply:
Yeah, it's probably something like that :+1: